src/EditorFeatures/Core/CodeDefinitionWindow/DefinitionContextTracker.cs

```diff
@@ -28,181 +29,199 @@
-            }
-            else if (_metadataAsSourceFileService.IsNavigableMetadataSymbol(symbol))
-            {
-                var options = _globalOptions.GetMetadataAsSourceOptions(document.Project.Services);
-                var declarationFile = await _metadataAsSourceFileService.GetGeneratedFileAsync(workspace, document.Project, symbol, signaturesOnly: false, options, cancellationToken).ConfigureAwait(false);
-                var identifierSpan = declarationFile.IdentifierLocation.GetLineSpan().Span;
-                return ImmutableArray.Create(new CodeDefinitionWindowLocation(symbol.ToDisplayString(), declarationFile.FilePath, identifierSpan.Start));
-            }
+        var symbolNavigationService = workspace.Services.GetRequiredService<ISymbolNavigationService>();
+        var definitionItem = symbol.ToNonClassifiedDefinitionItem(document.Project.Solution, includeHiddenLocations: false);
+        var result = await symbolNavigationService.GetExternalNavigationSymbolLocationAsync(definitionItem, cancellationToken).ConfigureAwait(false);
 
-            return ImmutableArray<CodeDefinitionWindowLocation>.Empty;
+        if (result != null)
+        {
+            return ImmutableArray.Create(new CodeDefinitionWindowLocation(symbol.ToDisplayString(), result.Value.filePath, result.Value.linePosition));
         }
+        else if (_metadataAsSourceFileService.IsNavigableMetadataSymbol(symbol))
+        {
+            var options = _globalOptions.GetMetadataAsSourceOptions(document.Project.Services);
+            var declarationFile = await _metadataAsSourceFileService.GetGeneratedFileAsync(workspace, document.Project, symbol, signaturesOnly: false, options, cancellationToken).ConfigureAwait(false);
+            var identifierSpan = declarationFile.IdentifierLocation.GetLineSpan().Span;
+            return ImmutableArray.Create(new CodeDefinitionWindowLocation(symbol.ToDisplayString(), declarationFile.FilePath, identifierSpan.Start));
+        }
+
+        return ImmutableArray<CodeDefinitionWindowLocation>.Empty;
+    }
+
+    private static async Task<ImmutableArray<INavigableItem>> GetDefinitionsAsync(Document document, int position, CancellationToken cancellationToken)
+    {
+        // Try IFindDefinitionService first. Until partners implement this, it could fail to find a service, so fall back if it's null.
+        var findDefinitionService = document.GetLanguageService<IFindDefinitionService>();
+        if (findDefinitionService != null)
+            return await findDefinitionService.FindDefinitionsAsync(document, position, cancellationToken).ConfigureAwait(false);
+
+        // Removal of this codepath is tracked by https://github.com/dotnet/roslyn/issues/50391. Once it is removed,
+        // this GetDefinitions method should be inlined into call sites.
+        var goToDefinitionsService = document.GetLanguageService<IGoToDefinitionService>();
```

Comment:
this changed to be optional.

Reply:
Hmm, given there's a tracking bug, is this just old? Can we just delete it now?